modules/extras/src/main/scala/io/circe/extras/Extras.scala

```diff
@@ -0,0 +1,84 @@
+package io.circe.extras
+
+import io.circe.{ Json, JsonNumber, JsonObject }
+
+import scala.collection.immutable.{ Map, Set }
+
+object Extras {
+
+  /**
+   * For a [[Json]], sanitize any [[JsonObject]]'s key's values if
+   * @keysToSanitize contains it.
+   *
+   * The original author's motivation for this function should explain why someone may want to use it.
+   *
+   * When the author had defined an API Integration, namely an HTTP Client talking to a 3rd Party API, he needed
+   * a way to log the full HTTP Response's JSON Payload. However, the entire JSON could not be logged since
+   * it contained sensitive information. So, that resulted in the author writing a function for
+   * sanitizing JSON, i.e. apply a function: [[Json]] => [[Json]] that replaces sensitive keys' values with
+   * dummy values, e.g. "XXXX". A set of keys' values serves as a whitelist.
```

Comment:
I'm glad that I wrote this comment. `keysToSanitize` should actually be `whitelist`. In other words, my implementation is wrong.

I'm going to update it now.

Reply:
Done via https://github.com/circe/circe/pull/1436#commits-pushed-33f63b5